src/auth.test.ts

```diff
@@ -0,0 +1,219 @@
+
+// Setup BlueButton class instance
+const CLIENT_ID = "foo";
+const CLIENT_SECRET = "bar";
+const CALLBACK_URL = "http://localhost/callback/";
+const bb = new BlueButton({
+  clientId: CLIENT_ID,
+  clientSecret: CLIENT_SECRET,
+  callbackUrl: CALLBACK_URL,
+});
+
+// TEST CONSTANTS
+const BASE_AUTH_URL: string =
+  "https://sandbox.bluebutton.cms.gov/2/o/authorize";
+
+describe("auth method generateAuthData()", () => {
+  const AuthData = bb.generateAuthData();
+
+  it("expect codeChallenge length != 0", () => {
+    expect(AuthData.codeChallenge).not.toHaveLength(0);
+  });
+
+  it("expect verifier length != 0", () => {
+    expect(AuthData.verifier).not.toHaveLength(0);
+  });
+
+  it("expect state length != 0", () => {
+    expect(AuthData.state).not.toHaveLength(0);
+  });
+});
+
+test("expect auth method generateAuthorizeUrl()", () => {
+  const AuthData = bb.generateAuthData();
+
+  const expectedUrl: string =
+    `${BASE_AUTH_URL}?client_id=${CLIENT_ID}` +
+    `&redirect_uri=${CALLBACK_URL}&state=${AuthData.state}` +
+    `&response_type=code&code_challenge_method=S256&code_challenge=${AuthData.codeChallenge}`;
+
+  const url = bb.generateAuthorizeUrl(AuthData);
+
+  expect(url).toBe(expectedUrl);
+});
+
+test("expect auth method generateTokenPostData() function", () => {
+  const AuthData = bb.generateAuthData();
+
+  const expectedPostData = {
+    client_id: "foo",
+    client_secret: "bar",
+    code: "test-code",
+    code_challenge: AuthData.codeChallenge,
+    code_verifier: AuthData.verifier,
+    grant_type: "authorization_code",
+    redirect_uri: "http://localhost/callback/",
+  };
+
+  const postData = generateTokenPostData(bb, AuthData, "test-code");
+
+  expect(postData).toStrictEqual(expectedPostData);
+});
+
+describe("auth method getAuthorizationToken", () => {
+  const authData = bb.generateAuthData();
+
+  const BB2_ACCESS_TOKEN_URL = bb.baseUrl + "/" + bb.version + "/o/token/";
+
+  const mockResponse = {
+    status: 200,
+    data: {
+      access_token: "66ClP4JCjpdxmkC6bEKYQFLOWXnraJ",
+      expires_in: 36000,
+      token_type: "Bearer",
+      scope: [
+        "introspection",
+        "patient/Coverage.read",
+        "patient/ExplanationOfBenefit.read",
+        "patient/Patient.read",
+        "profile",
+      ],
+      refresh_token: "jV3knA4xmZ1enK8Rg1Lub5hmIsc9Ad",
+      patient: "-20140000000051",
+      expires_at: 1646195004.3654683,
+    },
+  };
+
+  const mockResponseMissingExpiresAt = {
+    status: 200,
+    data: {
+      access_token: "66ClP4JCjpdxmkC6bEKYQFLOWXnraJ",
+      expires_in: 36000,
+      token_type: "Bearer",
+      scope: [
+        "introspection",
+        "patient/Coverage.read",
+        "patient/ExplanationOfBenefit.read",
+        "patient/Patient.read",
+        "profile",
+      ],
+      refresh_token: "jV3knA4xmZ1enK8Rg1Lub5hmIsc9Ad",
+      patient: "-20140000000051",
+    },
+  };
+
+  it("expect successful response", async () => {
```

Comment:
```suggestion
  it("expect successful response", async () => {
    mockedAxios.post.mockResolvedValueOnce(mockResponse);
    const expectedAuthToken = new AuthorizationToken(mockResponse.data);
    const ret = await bb.getAuthorizationToken(
        authData,
        "test-code",
        authData.state,
        "test-error"
      );
      expect(axios.post).toHaveBeenCalledWith(
      BB2_ACCESS_TOKEN_URL,
      expect.anything()
    );
    // write some test to validate `ret`

Reply:
Since you have the `async` at the top, I don't think you need a separate block here. If an error is thrown during a test, the test will fail. It'll also just tell you what error happened.

Reply:
@nbragdon This works. I think did this when trying to get mocks to work. TY!